File: Chapter 04/C Code/4_13.c
#include<stdio.h>
#define min(n,m) n<m?n:m

void main()
{
int n,m,hcf,i;
printf("Enter 2 numbers: ");
scanf("%d%d", &n, &m);

hcf=1;
for (i=min(n,m); i>1; i--)
{
if (n>=i  && m >=i)
{
if (n%i ==0 && m%i == 0)
{
n=n/i;
m=m/i;
hcf=hcf*i;
}
}
else
break;
}
printf("HCF is %d", hcf);
}

File: Chapter 04/Python code/4_13.py
n=int(input("Enter 1st number: "))
m=int(input("Enter 2nd number: "))
hcf=1
for i in range(min(n,m), 1, -1):
    if n>=i and m >=i:
        if n%i ==0 and m%i == 0:
            n=n/i
            m=m/i
            hcf=hcf*i
    else:
        break
print(hcf)

